location/motif.py

```diff
@@ -833,6 +833,129 @@ def compute_nodes(df,
     return df, nodes
 
 
+def generate_motifs(data,
+                    nodes,
+                    sr_col='stay_region',
+                    round_trip=True,
+                    insert_home=True,
+                    home=None,
+                    dayofweek=[0, 1, 2, 3, 4],
+                    trav_dist_th=50000,
+                    valid_timeslot_th=8):
+    """
+    Generate moitfs for given data.
+
```

Comment:
I'd highly recommend either explaining your algorithm here or point to a reference.